spinalcordtoolbox/scripts/sct_maths.py

```diff
@@ -479,16 +481,18 @@ def main(argv: Sequence[str]):
                         list_data = get_data_arrays(data.shape, arg_value)
                     except ValueError as e:
                         printv(f"ERROR: -{arg_name}: {e}", 1, 'error')
-                    # the multiplication can't use *= because the dtypes could be different
-                    data = data * np.prod(list_data, axis=0)
+                    # for multiplication, this dtype is usually ok
+                    safe_dtype = np.result_type(data, *list_data)
+                    data = np.multiply(data, np.prod(list_data, axis=0, dtype=safe_dtype), dtype=safe_dtype)
 
             elif arg_name == "div":
                 try:
                     list_data = get_data_arrays(data.shape, arg_value)
                 except ValueError as e:
                     printv(f"ERROR: -{arg_name}: {e}", 1, 'error')
-                # the division can't use /= because the dtypes could be different
-                data = data / np.prod(list_data, axis=0)
+                # for division, make sure the dtype is at least floating point by including float16
+                safe_dtype = np.result_type(data, *list_data, np.float16)
```

Comment:
Tested this (pushing a test soon), and found it runs into: 

- https://github.com/spinalcordtoolbox/spinalcordtoolbox/issues/333